contrib/pmsnare/pmsnare.c

```diff
@@ -95,79 +97,97 @@ CODESTARTparse
 		--lenMsg;
 		++p2parse;
 	}
-	dbgprintf("pmsnare: separator [%d]'%s'  msg after the first separator: [%d]'%s'\n", tablength,
-	TabRepresentation,lenMsg, p2parse);
-	if ((lenMsg > tablength) && (*p2parse == '\t' || strncasecmp((char*) p2parse, TabRepresentation ,
-	tablength-1) == 0)) {
-	//if ((lenMsg > tablength) && (*p2parse == '\t' || *p2parse == '#')) {
-	dbgprintf("pmsnare: tab separated message\n");
-		if(strncasecmp((char*) (p2parse + tablength - 1), "MSWinEventLog", 13) == 0) {
-			snaremessage=13; /* 0 means not a snare message, a number is how long the tag is */
+	dbgprintf("pmsnare: separator [%d]'%s'  msg after the first separator: "
+			"[%d]'%s'\n", tablength,TabRepresentation,lenMsg, p2parse);
+	/* Is the first separator a tab? */
+	if ((lenMsg > tablength) && (*p2parse == '\t' || \
+			strncasecmp( \
+				(char*) p2parse, TabRepresentation , tablength-1) == 0)) {
+		dbgprintf("pmsnare: tab separated message\n");
+		
+		/* Measure length of tag. */
+		if(strncasecmp( \
+				(char*)(p2parse + tablength - 1), "MSWinEventLog", 13) == 0) {
+			len_snare_tag=13;
 		}
-		if(strncasecmp((char*) (p2parse + tablength - 1), "LinuxKAudit", 11) == 0) {
-			snaremessage=11; /* 0 means not a snare message, a number is how long the tag is */
+		if(strncasecmp( \
+				(char*) (p2parse + tablength - 1), "LinuxKAudit", 11) == 0) {
+			len_snare_tag=11; 
 		}
-		if(snaremessage) {
-			/* replace the tab with a space and if needed move the message portion up by the length of
-			TabRepresentation -2 characters to overwrite the extra : */
-			*p2parse = ' ';
-			lenMsg -=(tablength-2);
-			p2parse++;
-			lenMsg--;
-			memmove(p2parse, p2parse + (tablength-2), lenMsg);
-			*(p2parse + lenMsg) = '\n';
-			*(p2parse + lenMsg + 1)  = '\0';
-			pMsg->iLenRawMsg -=(tablength-2);
-			pMsg->iLenMSG -=(tablength-2);
-			p2parse += snaremessage;
-			lenMsg -= snaremessage;
-			*p2parse = ' ';
-			p2parse++;
-			lenMsg--;
-			lenMsg -=(tablength-2);
-			memmove(p2parse, p2parse + (tablength-2), lenMsg);
-			*(p2parse + lenMsg) = '\n';
-			*(p2parse + lenMsg + 1)  = '\0';
-			pMsg->iLenRawMsg -=(tablength-2);
-			pMsg->iLenMSG -=(tablength-2);
-			dbgprintf("found a Snare message with snare not set to send syslog messages\n");
+		
+		/* Found a Snare event (not syslog). */
+		if(len_snare_tag) {
+			dbgprintf("found a Snare message with snare not set to send "
+					"syslog messages\n");
+			/* Replace tab with space. */
+			if (*p2parse == '\t') {
+				*p2parse = ' ';
+			} else if (*p2parse == '#') {
+				/* If character is escaped, we need to shift everything over. */
+				*p2parse = ' ';
+				p2parse++;
+				lenMsg--;
+				/* Length of tab character minus char already replaced and
+				 * null terminator */
+				lenMsg -= (tablength-2);	
+				memmove(p2parse, p2parse+(tablength-2), lenMsg);
+				*(p2parse + lenMsg) = '\n';
+				*(p2parse + lenMsg + 1)  = '\0';
+				pMsg->iLenRawMsg -=(tablength-2);
```

Comment:
Yes, it seems to work well for that specific use case and it is likely the most common. I hope that my changes will expand the functionality so that it can be used in many other scenarios regardless of other config settings.